nomad/csi_endpoint.go

```diff
@@ -1062,20 +1109,37 @@ func (v *CSIVolume) Create(args *structs.CSIVolumeCreateRequest, reply *structs.
 	// eval" that can do the plugin RPCs async.
 
 	var mErr multierror.Error
+	var index uint64
 
 	for _, valid := range validatedVols {
-		err = v.createVolume(valid.vol, valid.plugin)
-		if err != nil {
-			multierror.Append(&mErr, err)
+		if valid.current != nil {
+			// reconcile mutable fields
+			cp := valid.current.Copy()
+			err = v.reconcileVolume(valid.plugin, cp, valid.vol)
+			if err != nil {
+				multierror.Append(&mErr, err)
```

Comment:
`mErr` is a multierror, so we can get away with using the builtin append here which is more efficient for both time and complexity. This also applies to the lines 1129 and 1141 below.
```suggestion
				mErr.Errors = append(mErr.Errors, err)
```

Reply:
I really considered using the (relatively) new stdlib [`errors.Join()`](https://pkg.go.dev/errors#Join) instead, but honestly multierror's error string is still easier to read, IMO.  stdlib's just joins them together with newlines.